tests/hrex/test_hrex.py

```diff
@@ -40,3 +44,105 @@ def test_get_samples_by_iter_by_replica_invalid_args():
         get_samples_by_iter_by_replica([[1], [2, 3]], [[ReplicaIdx(0)], [ReplicaIdx(0)]])
     with pytest.raises(AssertionError):
         get_samples_by_iter_by_replica([[1, 2], [3, 4]], [[ReplicaIdx(0)], [ReplicaIdx(0), ReplicaIdx(1)]])
+
+
+def simulate_perfect_mixing_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """assume every step of HREX perfectly mixed all replicas"""
+    rng = np.random.default_rng(num_states)
+    inds = np.arange(num_states)
+    traj = []
+    for _ in range(num_frames):
+        rng.shuffle(inds)
+        traj.append(np.array(inds).tolist())
+    return traj
+
+
+def simulate_slow_mixing_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """assume every step of HREX only succeeds in making K nearest-neighbor swaps"""
+    traj = [np.arange(num_states).tolist()]
+    for _ in range(num_frames - 1):
+        current_state = np.array(traj[-1])
+        for _ in range(num_states):
+            i = np.random.randint(num_states - 1)
+            j = i + 1
+
+            x_i = current_state[i]
+            x_j = current_state[j]
+
+            current_state[j] = x_i
+            current_state[i] = x_j
+
+        traj.append(current_state.tolist())
+    return traj
+
+
+def simulate_bottlenecked_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """simulate_slow_mixing_hrex, but there's a state near K/2 that never swaps with neighbors"""
```

Comment:
`K` -> `num_states`

Reply:
Fixed up in https://github.com/proteneer/timemachine/pull/1314/commits/68079aab2b6b8fdcef0f050a75313708516ba6a3